client/layout/masterbar/logged-in.jsx

```diff
@@ -837,7 +837,7 @@ export default connect(
 		// by the user yet
 		const currentSelectedSiteId = getSelectedSiteId( state );
 		const siteId =
-			currentSelectedSiteId || getPrevSelectedSiteId( state ) || getPrimarySiteId( state );
+			currentSelectedSiteId || getMostRecentlySelectedSiteId( state ) || getPrimarySiteId( state );
```

Comment:
With the new changes here to most recent, I think we could remove the `currentSelectedSiteId` check from this altogether? If it has a truthy value, mostRecent will always have the same value now anyways.